docs/docs/saving_and_loading_from_catalog.rst

```diff
@@ -56,28 +56,40 @@ To use catalog objects, simply specify their name in the Unitxt object that will
 Modifying Catalog Assets on the Fly
 -----------------------------------
 
-To modify a catalog asset's fields dynamically, use the syntax: `asset.name[key_to_modify=new_value]`. To assign lists, use: `asset.name[key_to_modify=[new_value_0, new_value_1]]`. For instance, to change the metric list of a task:
+To modify a catalog asset's fields dynamically, upon fetching the asset from the catalog, and instantiating it into a python object of type ``unitxt.Artifact``, use the syntax: ``asset.name[key_to_modify=new_value]``. 
+To assign lists, use: ``asset.name[key_to_modify=[new_value_0, new_value_1]]``. 
+To assign dictionaries, use: ``asset.name[key_to_modify={new_key_0=new_value_0,new_key_1=new_value_1}]``.
+Note that the whole new value of the field has to be specified; not just one item of a list, or one key of the dictionary.
+For instance, to change the metric list and the reference specification of a task:
 
 .. code-block:: python
 
     from unitxt.card import TaskCard
 
     card = TaskCard(
         ...
-        task="tasks.my_task[metrics=[metrics.accuracy, metrics.f1[reduction=median]]]"
+        task="tasks.my_task[metrics=[metrics.accuracy, metrics.f1[reduction=median]],reference_fields={output=int}]"
     )
 
 Accessing Catalog Assets Directly
 ---------------------------------
 
-Use `get_from_catalog` to directly access catalog assets:
+Use ``get_from_catalog`` to directly access catalog assets, and obtain an asset instantiated as a python object of type ``unitxt.Artifact``:
 
 .. code-block:: python
 
     from unitxt import get_from_catalog
 
     my_task = get_from_catalog("tasks.my_task")
 
+A Catalog Asset Linking to Another Catalog Asset
+------------------------------------------------
+
+A catalog asset can be just a link to another asset. This feature comes handy when for some reason, ``asset1`` -- the name of an asset, which reflects its place in the catalog, is changed to ``asset2``, while much code already exists where the old name of the asset, ``asset1`` is hard coded.
```

Comment:
```suggestion
A catalog asset can be just a link to another asset. This feature comes handy when for some reason, we want to change the catalog name of an existing asset, while there is already code that uses the old name of the asset and we want to avoid non-backward compatible changes.
```

Reply:
took your rephrasing.